src/main/kotlin/org/rust/ide/annotator/fixes/AddModuleFileFix.kt

```diff
@@ -13,18 +13,36 @@ import com.intellij.psi.PsiFile
 import org.rust.ide.refactoring.RsPromoteModuleToDirectoryAction
 import org.rust.lang.core.psi.RsFile
 import org.rust.lang.core.psi.RsModDeclItem
-import org.rust.lang.core.psi.ext.getOrCreateModuleFile
 
 /**
  * Creates module file by the given module declaration.
  */
 class AddModuleFileFix(
     modDecl: RsModDeclItem,
-    private val expandModuleFirst: Boolean
+    private val expandModuleFirst: Boolean,
+    private val location: Location = Location.File
 ) : LocalQuickFixAndIntentionActionOnPsiElement(modDecl) {
-    override fun getText(): String = "Create module file"
+    private val text = run { "Create module file `${modDecl.getPath()}`" }
+    override fun getText() = text
+
+    override fun getFamilyName() = "Create module file"
+
+    enum class Location {
+        /**
+         * Creates a module in the same directory. E.g. `mod foo` creates a file `foo.rs`
+         */
+        File,
+        /**
+         * Creates a module in a subdirectory. E.g. `mod foo` creates a file `foo/mod.rs`
+         */
+        Directory
+    }
+
+    private fun RsModDeclItem.getPath(): String = when (location) {
```

Comment:
Maybe [it's better](https://kotlinlang.org/docs/reference/coding-conventions.html#functions-vs-properties) to create a property instead of the method?